crates/gitbutler-core/src/git/repository_ext.rs

```diff
@@ -26,6 +26,10 @@ pub trait RepositoryExt {
     fn in_memory<T, F>(&self, f: F) -> Result<T>
     where
         F: FnOnce(&git2::Repository) -> Result<T>;
+    fn integration_commit(&self) -> Result<git2::Commit<'_>>;
+    fn sign_buffer(&self, buffer: &CommitBuffer) -> Result<String>;
```

Comment:
To my mind, it's very important to (eventually) be able to add documentation to these methods. Here, it's unclear if it returns a signed version of `buffer`, or the signature. I'd lean towards thinking it's the signature, but I'd be confused by the `String` type which implies it might be hex-encoded.
That's too many questions for me.

Personally, I tend to document as precisely as possible what I know this method is supposed to do while implementing it, because that's when I am the expert. 2 weeks later, I won't be that anymore and happy about some text that explains the intent of what seemed so obvious a couple of weeks ago.

Reply:
Yep! String is a really bad type there.

I'm still not very happy with the whole CommitBuffer thing because it doesn't support parsing a commit that has been signed either.